spacy/cli/project/run.py

```diff
@@ -71,6 +74,12 @@ def project_run(
     commands = {cmd["name"]: cmd for cmd in config.get("commands", [])}
     workflows = config.get("workflows", {})
     validate_subcommand(list(commands.keys()), list(workflows.keys()), subcommand)
+
+    req_path = project_dir / "requirements.txt"
+    if os.path.exists(req_path):
+        with req_path.open() as requirements_file:
+            _validate_requirements([req.replace("\n", "") for req in requirements_file])
```

Comment:
We need some kind of mechanism to make this optional, or at least provide some option to a user to avoid these warnings. Imagine a project with multiple workflows where a `torch` dependency is only needed for `workflowB` and not `workflowA`- then you don't want to keep seeing these warnings over and over again if you're only running `workflowA`, and you don't want to "forced" into installing a dependency you don't really need.

Reply:
I hadn't considered that. Perhaps a `--no-validate` (assuming we want to validate dependencies by default) flag for `spacy project run`?

Reply:
I think a `project.yml` setting sounds better?

Reply:
So if `validate: False` is in `project.yml`, we don't validate; if `validate` is not in the config or `validate: True` is, we validate?

Reply:
That sounds like a good plan, but perhaps make the setting even more specific so it doesn't clash with other usages of "validating stuff" in a project. Something like `validate_requirements`?

Reply:
I think `validate` is overloaded here and might suggest something like `check_requirements` instead, which also sounds more like `pip check`.

Reply:
Both's good with me. I'll go with `check_requirements` then, if @svlandeg doesn't mind?

Reply:
Added.

Reply:
@adrianeboyd @svlandeg Ping for feedback on `check_requirements`.